src/org/zaproxy/zap/extension/ascan/ActiveScan.java

```diff
@@ -289,7 +296,27 @@ public void run() {
             });
         }
 	}
-	
+
+    private void removeHistoryReference(HistoryReference remRef) {
+        if (View.isInitialised()) {

```

Comment:
So they should always be using the messagesTableModel add and remove methods not the local ones?